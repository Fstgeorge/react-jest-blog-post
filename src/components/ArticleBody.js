import React, { PropTypes } from 'react';

const ArticleBody = ({ body }) => (
  <div>
    <p>{body}</p>
  </div>
);

ArticleBody.propTypes = {
  body: PropTypes.string.isRequired,
}

export default ArticleBody;
