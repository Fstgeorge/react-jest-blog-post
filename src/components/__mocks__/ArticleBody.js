import React from 'react';

const ArticleBody = props => (
  <span>ArticleBody {JSON.stringify(props)}</span>
);

module.exports = ArticleBody;
