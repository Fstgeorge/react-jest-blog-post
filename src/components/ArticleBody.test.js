import renderer from 'react-test-renderer';
import React from 'react';
import ArticleBody from './ArticleBody';

test('ArticleTest renders correctly', () => {
  const tree = renderer.create(
    <ArticleBody body="A title" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
