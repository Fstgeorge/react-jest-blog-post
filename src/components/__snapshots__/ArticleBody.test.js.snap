exports[`test ArticleTest renders correctly 1`] = `
<div>
  <p>
    A title
  </p>
</div>
`;
